### PR TITLE
us12 backend

### PR DESCRIPTION
**Pull request by thomasmckennatsb**

### CODE REVIEW

File: src/main/java/com/kainos/ea/controller/WebService.java [hunk 1]
```diff
@@ -141,4 +141,26 @@ public Response getBandCompetency(@PathParam("id") int band_id)
...
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createJobRole(JobRoleRequest jobRoleReq){
+        try{
+            if (jobRoleValidator.isValidJobRole(jobRoleReq)){
+                try {
+                    int id = rolesService.createJobRole(jobRoleReq);
+                    return Response.status(HttpStatus.CREATED_201).entity(id).build();
+                } catch (Exception | DatabaseConnectionException e) {
+                    e.printStackTrace();
```
Comment (bannister-samuel) [inline src/main/java/com/kainos/ea/controller/WebService.java]: I don't think you should be printing the error

File: src/main/java/com/kainos/ea/controller/WebService.java [hunk 2]
```diff
@@ -141,4 +141,26 @@ public Response getBandCompetency(@PathParam("id") int band_id)
...
+                    return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500).build();
+                }
+            } else {
+                return Response.status(HttpStatus.BAD_REQUEST_400).build();
+            }
+        } catch (UrlNotValidException e) {
+            return Response.status(HttpStatus.BAD_REQUEST_400).build();
+        }
+    }
 }
```
Comment (bannister-samuel) [inline src/main/java/com/kainos/ea/controller/WebService.java:165]: Blank space at end of file

File: src/main/java/com/kainos/ea/dao/RolesDao.java [hunk 3]
```diff
@@ -114,4 +114,30 @@ public JobSpecification getAllSpecification(Connection c, int role_id) throws SQ
...
+
+            while (rs.next()) {
+                newId = rs.getInt(1);
+            }
+        } catch (SQLException e) {
+            System.out.println(e.getMessage());
+        }
+        return newId;
+    }
 }
```
Comment (bannister-samuel) [inline src/main/java/com/kainos/ea/dao/RolesDao.java:143]: Blank space at end of file

**Review by bannister-samuel — CHANGES_REQUESTED** [review]
see comments - not 100% sure if the whitespace is required at EOF

File: src/main/java/com/kainos/ea/controller/WebService.java [hunk 4]
```diff
@@ -141,4 +141,25 @@ public Response getBandCompetency(@PathParam("id") int band_id)
...
+                    return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500).build();
+                }
+            } else {
+                return Response.status(HttpStatus.BAD_REQUEST_400).build();
+            }
+        } catch (UrlNotValidException e) {
+            return Response.status(HttpStatus.BAD_REQUEST_400).build();
+        }
+    }
+}
```
Comment (boydstepnew) [inline src/main/java/com/kainos/ea/controller/WebService.java]: Stick some newlines in where you see this symbol (check the other files too)

File: src/main/java/com/kainos/ea/dao/RolesDao.java [hunk 5]
```diff
@@ -114,4 +114,30 @@ public JobSpecification getAllSpecification(Connection c, int role_id) throws SQ
             throw new SQLException ("Select query for job specifications is incorrect");
         }
     }
-}
+
+    public int createJobRole(JobRoleRequest jobRoleXL, Connection c) throws SQLException{
+
+        int newId = 0;
```
Comment (boydstepnew) [inline src/main/java/com/kainos/ea/dao/RolesDao.java]: Hm, this name took me a second to understand - maybe something like "createdRoleId" would be clearer?

File: src/main/java/com/kainos/ea/dao/RolesDao.java [hunk 6]
```diff
@@ -114,4 +114,30 @@ public JobSpecification getAllSpecification(Connection c, int role_id) throws SQ
...
+        try {
+            String updateQuery = "INSERT INTO job_role (band_id, job_family_id, kainos_job_title, job_specification, job_spec_link) VALUES (?,?,?,?,?)";
+
+            PreparedStatement prepStm = c.prepareStatement(updateQuery, Statement.RETURN_GENERATED_KEYS);
+            prepStm.setInt(1, jobRoleXL.getBandId());
```
Comment (boydstepnew) [inline src/main/java/com/kainos/ea/dao/RolesDao.java]: I thought JobRoleXL was removed?

File: src/main/java/com/kainos/ea/dao/RolesDao.java [hunk 7]
```diff
@@ -114,4 +114,30 @@ public JobSpecification getAllSpecification(Connection c, int role_id) throws SQ
             throw new SQLException ("Select query for job specifications is incorrect");
         }
     }
-}
+
+    public int createJobRole(JobRoleRequest jobRoleXL, Connection c) throws SQLException{
```
Comment (boydstepnew) [inline src/main/java/com/kainos/ea/dao/RolesDao.java]: Ah, there's the issue - this variable needs renamed

**Review by boydstepnew — APPROVED** [review]

**Review by bannister-samuel — APPROVED** [review]

**Pull request merged**